sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/AbstractScheduler.java

```diff
@@ -23,14 +29,28 @@
     // Mesos may call registered() multiple times in the lifespan of a Scheduler process, specifically when there's
     // master re-election. Avoid performing initialization multiple times, which would cause resourcesQueue to be stuck.
     private final AtomicBoolean isAlreadyRegistered = new AtomicBoolean(false);
+    protected final OfferQueue offerQueue = new OfferQueue();
     protected SchedulerDriver driver;
     protected DefaultReconciler reconciler;
+    private Set<Protos.OfferID> offersInProgress = new HashSet<>();
+
+    /**
+     * Executor for handling TaskStatus updates in {@link #statusUpdate(SchedulerDriver, Protos.TaskStatus)}.
+     */
+    protected final ExecutorService statusExecutor = Executors.newSingleThreadExecutor();
```

Comment:
It looks to me that ultimately `TaskStatus` messages end up in the `update` call of `DeploymentStep` in all the `DefaultPlanCoordinator` case.  This updates only the status member variable which whose access is synchronized.  Will keep looking though.  Tests are not passing pretty broadly.  Trying to get a local repro to see what's happening.